### ISSUE TITLE
Ring's heartbeat too much times...

### ISSUE THREAD

**Issue by hnlq715** [go-redis/redis#349]
As we can see, there were too much heartbeat check request to redis...
Maybe we should change this ticker timer to `1 * time.Second`?
```
// heartbeat monitors state of each shard in the ring.
func (ring *Ring) heartbeat() {
	ticker := time.NewTicker(100 * time.Millisecond)
	defer ticker.Stop()
```

```
17:15:25.738588 IP 192.168.1.244.27075 > 192.168.1.29.webcache: Flags [S], seq 2251986615, win 14600, options [mss 1460,sackOK,TS val 475135508 ecr 0,nop,wscale 9], length 0
17:15:25.738717 IP 192.168.1.29.webcache > 192.168.1.244.27075: Flags [R.], seq 0, ack 2251986616, win 0, length 0
17:15:25.789302 IP 192.168.1.244.27076 > 192.168.1.29.webcache: Flags [S], seq 1675077536, win 14600, options [mss 1460,sackOK,TS val 475135559 ecr 0,nop,wscale 9], length 0
17:15:25.789495 IP 192.168.1.29.webcache > 192.168.1.244.27076: Flags [R.], seq 0, ack 1675077537, win 0, length 0
17:15:25.811380 IP 192.168.1.244.27077 > 192.168.1.29.webcache: Flags [S], seq 3767825438, win 14600, options [mss 1460,sackOK,TS val 475135581 ecr 0,nop,wscale 9], length 0
17:15:25.811568 IP 192.168.1.29.webcache > 192.168.1.244.27077: Flags [R.], seq 0, ack 3767825439, win 0, length 0
17:15:25.838568 IP 192.168.1.244.27087 > 192.168.1.29.webcache: Flags [S], seq 2920550697, win 14600, options [mss 1460,sackOK,TS val 475135608 ecr 0,nop,wscale 9], length 0
17:15:25.838767 IP 192.168.1.29.webcache > 192.168.1.244.27087: Flags [R.], seq 0, ack 2920550698, win 0, length 0
17:15:25.889340 IP 192.168.1.244.27088 > 192.168.1.29.webcache: Flags [S], seq 3343803042, win 14600, options [mss 1460,sackOK,TS val 475135659 ecr 0,nop,wscale 9], length 0
17:15:25.889497 IP 192.168.1.29.webcache > 192.168.1.244.27088: Flags [R.], seq 0, ack 3343803043, win 0, length 0
17:15:25.911179 IP 192.168.1.244.27089 > 192.168.1.29.webcache: Flags [S], seq 4112135465, win 14600, options [mss 1460,sackOK,TS val 475135681 ecr 0,nop,wscale 9], length 0
17:15:25.911405 IP 192.168.1.29.webcache > 192.168.1.244.27089: Flags [R.], seq 0, ack 4112135466, win 0, length 0
17:15:25.938556 IP 192.168.1.244.27092 > 192.168.1.29.webcache: Flags [S], seq 1532826696, win 14600, options [mss 1460,sackOK,TS val 475135708 ecr 0,nop,wscale 9], length 0
17:15:25.938750 IP 192.168.1.29.webcache > 192.168.1.244.27092: Flags [R.], seq 0, ack 1532826697, win 0, length 0
17:15:25.989313 IP 192.168.1.244.27093 > 192.168.1.29.webcache: Flags [S], seq 3533341137, win 14600, options [mss 1460,sackOK,TS val 475135759 ecr 0,nop,wscale 9], length 0
17:15:25.989467 IP 192.168.1.29.webcache > 192.168.1.244.27093: Flags [R.], seq 0, ack 3533341138, win 0, length 0
```

**Comment by vmihailenco** [comment]
TBH I don't understand the dump you posted :) But I will make heartbeat configurable and change it to 500ms.

**Comment by vmihailenco** [comment]
Added `HearbeatFrequency = 500ms` and lowered threshold to 3. Released in v4.1.2.
